proxy/envoy/discovery.go

```diff
@@ -321,33 +317,33 @@ func (ds *DiscoveryService) clearCache() {
 	ds.rdsCache.clear()
 }
 
-// ListAllEndpoints responds with all Services and is restricted to a single service key
+// ListAllEndpoints responds with all Services and is not restricted to a single service-key
 func (ds *DiscoveryService) ListAllEndpoints(request *restful.Request, response *restful.Response) {
 
-	var servicesArray []*service
-	for _, svc := range ds.services.Services() {
-		for _, port := range svc.Ports {
+	var services []*keyAndService
```

Comment:
Extra care should be taken when declaring `services` and `hosts` since they will be encoded as JSON objects. See https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices.

Any one of the following variants should work:
```golang
hosts := []*host
hosts := make([]*host)
hosts := make([]*host, 0, len(instances))
```